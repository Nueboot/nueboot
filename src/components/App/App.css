html {
  height: 100%;
  width: 100%;
  overflow: hidden;
}

body {
  height: 100%;
  padding: 0;
  overflow: auto;
  margin: 0;
  -webkit-overflow-scrolling: touch;
}

body {
  font-family: 'Heebo', 'Helvetica', 'Arial', sans-serif;
  color: '#999';
}

input {
  border: none;
  border-width: none;
}

input:focus {
  border-bottom-width: 2px;
  border-bottom-color: black;
  outline: none;
}

input::placeholder {
  font-family: 'Heebo';
  color: #999;
}

html, body, #root {
  width: 100%;
  height: 100%;
  margin: 0;
}

.app {
  align-content: center;
  min-height: 100%;
  display: grid;
  grid-template-rows: auto 1fr auto;
}