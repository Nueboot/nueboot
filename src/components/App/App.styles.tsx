import styled from "styled-components";

export const Container = styled.div`
  margin: 0 auto;
  max-width: 960px;
  margin-top: 60px;

  @media (max-width: 960px) {
    margin: 60px 20px;
  }
`;
