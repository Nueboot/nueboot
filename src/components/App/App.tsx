import createHistory from 'history/createBrowserHistory';
import React, { Component } from 'react';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'react-router-redux';

import configureStore from '../../store/configureStore';
import Routes from '../Routes';

const history = createHistory();â€‹
const store = configureStore(history);

(window as any).store = store;

export default class App extends Component {
  public render() {
    return (
      <Provider store={store}>
        <ConnectedRouter history={history}>
          <Routes />
        </ConnectedRouter>
      </Provider>
    );
  }
}
