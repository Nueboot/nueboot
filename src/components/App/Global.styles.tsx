import "typeface-heebo";
import { createGlobalStyle } from "styled-components";

const GlobalStyle = createGlobalStyle`
  body {
    font-family: 'Heebo';
    margin: 0;
    padding: 0;
    height: 100%;
    width: 100%;
  }

  a {
    color: black;
    text-decoration: none;
  }

`;

export default GlobalStyle;
