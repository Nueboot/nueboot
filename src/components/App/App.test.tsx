import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import { App, AppProps } from './App';

describe('App', () => {
  let component: ShallowWrapper<AppProps>;

  beforeEach(() => {
      component = shallow(<App authUser={null} />);
    });

  it('renders correctly', () => {
      expect(component).toMatchSnapshot();
    });
});
