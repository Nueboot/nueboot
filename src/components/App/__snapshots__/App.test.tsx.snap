// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders as expected 1`] = `
<Suspense
  fallback={
    <div>
      Loading...
    </div>
  }
>
  <Memo(GlobalStyleComponent) />
  <styled.div>
    <Router />
  </styled.div>
  <Footer />
</Suspense>
`;
